### PR TITLE
Merge upstream/master into trunk

### PR DESCRIPTION
**Pull request by WunderBart**
Let's update this fork 🧑‍🏭 

### Testing instructions
`yarn build` should not throw errors

### CODE REVIEW

File: .dev/config/webpack.config.js [hunk 1]
```diff
@@ -13,10 +13,13 @@ module.exports = {
...
 
 	entry: {
-		'coblocks': path.resolve( process.cwd(), 'src/blocks.js' ),
+		coblocks: path.resolve( process.cwd(), 'src/blocks.js' ),
+
+		// Styles
 		'coblocks-editor': path.resolve( process.cwd(), 'src/styles/editor.scss' ),
 		'coblocks-style': path.resolve( process.cwd(), 'src/styles/style.scss' ),
 
+		// Front-End Scripts
```
Comment (WunderBart) [inline .dev/config/webpack.config.js:22]: Most of those should also be removed but let's leave that for a follow-up PR. cc @fullofcaffeine

Comment (WunderBart) [reply .dev/config/webpack.config.js:22]: https://github.com/Automattic/coblocks/pull/8

**Review by fullofcaffeine — APPROVED** [review]
@WunderBart Thanks for working on this! ❤️ 

The build runs and finishes successfully. I've smoke tested it in a sandboxed simple site and seems to work fine. Bonus points: it also seems to work well on WoA (though I'm aware we won't be installing it in WoA now, though this opens the possibility to finally get rid of the divergence between versions).

**Comment by fullofcaffeine** [comment]
One thing that didn't work for me though was the WordPress `dev` environment. **Obviously, this is not a blocker for merging**, nor do I think we'll actually make much use of it, though I'd be curious to know if you also experienced problems with it @WunderBart. `yarn env install` fails with:

```
Warning: file_put_contents(/var/www/src/wp-config.php): failed to open stream: Permission denied in phar:///usr/local/bin/wp/vendor/wp-cli/config-command/src/Config_Command.php on line 193                                    
Error: Could not create new 'wp-config.php' file.                                                                                                                                                                               
child_process.js:836                                                                                                                                                                                                            
    throw err;                                                                                                                                                                                                                  
    ^

```

This causes databases connection issues in the WP instance started by `yarn dev start`, probably because the connection data is not available as the `wp-config.php` couldn't be copied over (for some reason).

Another thing is that at first glance, it seems to implement its own custom version of a `wp-env`-like tool 😕

**Comment by WunderBart** [comment]
> One thing that didn't work for me though was the WordPress `dev` environment. **Obviously, this is not a blocker for merging**, nor do I think we'll actually make much use of it, though I'd be curious to know if you also experienced problems with it @WunderBart. `yarn env install` fails with:
> 
> ```
> Warning: file_put_contents(/var/www/src/wp-config.php): failed to open stream: Permission denied in phar:///usr/local/bin/wp/vendor/wp-cli/config-command/src/Config_Command.php on line 193                                    
> Error: Could not create new 'wp-config.php' file.                                                                                                                                                                               
> child_process.js:836                                                                                                                                                                                                            
>     throw err;                                                                                                                                                                                                                  
>     ^
> ```
> 
> This causes databases connection issues in the WP instance started by `yarn dev start`, probably because the connection data is not available as the `wp-config.php` couldn't be copied over (for some reason).
> 
> Another thing is that at first glance, it seems to implement its own custom version of a `wp-env`-like tool 😕

Huh, I haven't experienced any problems with it so far. I think [it actually uses the original `wp-env`](https://wordpress.slack.com/archives/C03B0H5J0/p1630495198109800). Have you tried nuking everything with `yarn clean` or just `npx wp-env destroy` (if that's an option)?

**Pull request merged**